src/vm/costs/mod.rs

```diff
@@ -776,8 +826,12 @@ impl CostTracker for LimitedCostTracker {
         // grr, if HashMap::get didn't require Borrow, we wouldn't need this cloning.
         let lookup_key = (contract.clone(), function.clone());
         if let Some(cost_function) = self.contract_call_circuits.get(&lookup_key).cloned() {
-            let input_size = input.iter().fold(0, |agg, cur| agg + cur);
-            compute_cost(self, cost_function, input_size)?;
+            let input_sizes = if input.is_empty() {
```

Comment:
This shouldn't be necessary, we can just require that a cost-function that is targeting a zero-arg function have zero args as well.